src/frontend/mame/luaengine.cpp

```diff
@@ -1778,6 +1814,37 @@ void lua_engine::initialize()
 	image_type.set("is_reset_on_load", sol::property(&device_image_interface::is_reset_on_load));
 	image_type.set("must_be_loaded", sol::property(&device_image_interface::must_be_loaded));
 
+/*  device_slot_interface library
+ *
+ * manager:machine().slots[slot_name]
+ *
+ * slot.fixed - whether this slot is fixed, and hence not selectable by the user
+ * slot.has_selectable_options - does this slot have any selectable options at all?
+ * slot.default_option - returns the default option if one exists
+ * slot.options[] - get options table (k=name, v=device_slot_interface::slot_option)
+ */
+
+	auto slot_type = sol().registry().new_usertype<device_slot_interface>("slot", "new", sol::no_constructor);
+	slot_type["fixed"] = sol::property(&device_slot_interface::fixed);
+	slot_type["has_selectable_options"] = sol::property(&device_slot_interface::has_selectable_options);
+	slot_type["default_option"] = sol::property(&device_slot_interface::default_option);
+	slot_type["options"] = sol::property([](const device_slot_interface &slot) { return lua_engine::standard_tag_object_ptr_map<device_slot_interface::slot_option>(slot.option_list()); });
+
+
+/*  device_slot_interface::slot_option library
+ *
+ * manager:machine().slots[slot_name].options[option_name]
+ *
+ * slot_option.selectable - is this item selectable by the user?
+ * slot_option.default_bios - the default bios for this option
+ * slot_option.default_bios - the clock speed associated with this option
+ */
+
+	auto dislot_option_type = sol().registry().new_usertype<device_slot_interface::slot_option>("dislot_option", "new", sol::no_constructor);
+	dislot_option_type["selectable"] = sol::property(&device_slot_interface::slot_option::selectable);
+	dislot_option_type["default_bios"] = sol::property([](const device_slot_interface::slot_option &opt) { return opt.default_bios(); });
+	dislot_option_type["clock"] = sol::property([](const device_slot_interface::slot_option &opt) { return opt.clock(); });
```

Comment:
You can do this with a static cast rather than making a lambda wrapper – see e.g. `device_type["subdevice"]` and device_type["siblingdevice"]` which use static casts to choose between overloaded methods.